lib/dai-plugin-mcd/src/CdpType.js

```diff
@@ -14,6 +14,11 @@ export default class CdpType {
     this.currency = currency;
     this.ilk = ilk;
     this._ilkBytes = stringToBytes(this.ilk);
+    this._getPar();
```

Comment:
we're orphaning promises here. there's still the possibility of a race condition: we could start a contract call here, but just end up calling it again, if the second `_getPar` gets started before the first one finishes.

you could fix this by writing the getters like this instead:
```js
async _getPar() {
  if (!this._parPromise) this._parPromise = this._systemData.spot.par();
  return this._parPromise;
}
```
this way, no longer how many times you call the method, it only creates one promise. (well, i think there's still technically a race condition, but the window is much smaller.)

Reply:
hmm I see what you saying here, the example here is functionally the same to how it is written:
  async _getPar() {
    if (!this._par) this._par = await this._systemData.spot.par();
    return this._par;
  }
should I just rename it to have a promise so its clear?